src/domain.rs

```diff
@@ -1,10 +1,15 @@
-use crate::sync::atomic::{AtomicIsize, AtomicPtr, AtomicU64, AtomicUsize};
+use crate::sync::atomic::{AtomicIsize, AtomicPtr, AtomicU64, AtomicUsize, Ordering};
+use crate::{boxed::Box, marker::PhantomData};
 use crate::{Deleter, HazPtrRecord, Reclaim};
-use std::collections::HashSet;
-use std::marker::PhantomData;
-use std::sync::atomic::Ordering;
 
+#[cfg(not(feature = "std"))]
+use alloc::collections::BTreeSet as Set;
+#[cfg(feature = "std")]
+use std::collections::HashSet as Set;
```

Comment:
I'm inclined to just use `BTreeSet` universally here. What do you think?

Reply:
I've switched to it universally for now. A sorted Vec like what you mentioned would also be very interesting. Would love to see benchmarks comparing BTreeSet, HashSet and Vec!